### PR TITLE
DCR 3200 QnAMakerRecognizer improvements.

### PR DESCRIPTION
**Pull request by tomlm**
DCR #3200 
* Cleaned up CrossTrainedRecgonizerSet to collect recgonizer result by id instead of top intent name.
* added expressions to better configure QnAMakerRecognizer

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerRecognizer.cs [hunk 1]
```diff
@@ -96,6 +96,30 @@ public QnAMakerRecognizer()
...
 
+        /// <summary>
+        /// Gets or sets the whether to include the dialog name metadata for QnA context.
+        /// </summary>
+        /// <value>
+        /// A bool or boolean expression.
+        /// </value>
+        [DefaultValue(true)]
+        [JsonProperty("includeDialogNameInMetaData")]
+        public BoolExpression IncludeDialogNameInMetaData { get; set; } = true;
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerRecognizer.cs]: Casing: `Metadata` is a word on its own, so casing should be IncludeDialogNameInMetadata

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Pull request merged**